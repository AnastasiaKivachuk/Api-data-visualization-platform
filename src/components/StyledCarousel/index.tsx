import React from "react";
import styled from "@emotion/styled";

import ArrowCircleRightSharpIcon from "@mui/icons-material/ArrowCircleRightSharp";
import ArrowCircleLeftSharpIcon from "@mui/icons-material/ArrowCircleLeftSharp";
import Carousel from "react-material-ui-carousel";

type Props = {
    imagesUrls: string[],
    alt: string
}

const StyledCarousel = ({imagesUrls, alt}: Props): JSX.Element => {

    const StyledMuiCarousel = styled(Carousel)`
      width: 100%;
    `
    const WrapImg = styled.div`
      display: flex;
      justify-content: center;
    `

    const Img = styled.img`
      max-height: 400px;
    `

    return (
        <StyledMuiCarousel
            NextIcon={<ArrowCircleRightSharpIcon/>}
            PrevIcon={<ArrowCircleLeftSharpIcon/>}
        >
            {imagesUrls?.map((img) =>
                <WrapImg key={img}>
                    <Img src={img} alt={alt}/>
                </WrapImg>)}
        </StyledMuiCarousel>
    )
}

StyledCarousel.defaultProps = {
    alt: '',
    imagesUrls: []
}

export default StyledCarousel
