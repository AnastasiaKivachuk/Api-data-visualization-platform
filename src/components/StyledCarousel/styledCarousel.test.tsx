import {shallow} from "enzyme";

import StyledCarousel from "./index";


const images = ["https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg",
    "https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg",
    "https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png"]

const alt = "Thaicom 6";

const setUp = (props: { alt: string, imagesUrls: string[] } | {}) => shallow(<StyledCarousel {...props} />);

describe("StyledCarousel component", () => {
        it("should render StyledCarousel component", () => {
            const component = setUp({alt: alt, imagesUrls: images})
            expect(component).toMatchSnapshot();
        })

        it("should render StyledCarousel component without props", () => {
            const component = setUp({})
            expect(component).toMatchSnapshot();
        })

        it("check all elements", () => {
            const component = setUp({alt: alt, imagesUrls: images})
            expect(component.find('Styled(div)').length).toBe(3);
            expect(component.find('Styled(img)').length).toBe(3);
            const imgComponents = component.find('Styled(img)').map((item) => item.prop("src"))
            expect(imgComponents).toEqual(images)

        })

        it("check all elements with defaultProps", () => {
            const component = setUp({})
            expect(component.find('Styled(div)').length).toBe(0);
        })
    }
)