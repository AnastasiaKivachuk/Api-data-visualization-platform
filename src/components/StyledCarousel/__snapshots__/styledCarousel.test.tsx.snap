// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCarousel component should render StyledCarousel component 1`] = `
<Styled(Carousel)
  NextIcon={<UNDEFINED />}
  PrevIcon={<UNDEFINED />}
>
  <Styled(div)
    key="https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg"
  >
    <Styled(img)
      alt="Thaicom 6"
      src="https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg"
    />
  </Styled(div)>
  <Styled(div)
    key="https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg"
  >
    <Styled(img)
      alt="Thaicom 6"
      src="https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg"
    />
  </Styled(div)>
  <Styled(div)
    key="https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png"
  >
    <Styled(img)
      alt="Thaicom 6"
      src="https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png"
    />
  </Styled(div)>
</Styled(Carousel)>
`;

exports[`StyledCarousel component should render StyledCarousel component without props 1`] = `
<Styled(Carousel)
  NextIcon={<UNDEFINED />}
  PrevIcon={<UNDEFINED />}
/>
`;
